Automattic/wp-calypso
# Framework: Refactor away from _.assign()

#### Changes proposed in this Pull Request

Lodash's `assign()` is used many times in the codebase, but it's pretty straightforward to replace it with either spread or `Object.assign()`.  This PR updates all the instances.

If you want to find out why we're moving away from Lodash, see https://github.com/Automattic/wp-calypso/pull/50368 and p4TIVU-9Bf-p2.

#### Testing instructions

* Verify all tests pass: `yarn run test-client`, `yarn run test-packages` and `yarn run test-server`.
* Spend some time smoke testing Calypso and verify everything works alright.

Note that there are existing ESLint errors in some of the files we're touching here.
